### PR TITLE
Rewards distribution & Proxy

### PR DESCRIPTION
**Pull request by Amirjab21**

### CODE REVIEW

File: contracts/contracts/OpenSignal.sol [hunk 1]
```diff
@@ -48,12 +49,13 @@ contract OpenSignal is ERC2771Context, ReentrancyGuard {
         _;
     }
 
-    constructor(address _trustedForwarder, IERC20 _nativeToken) ERC2771Context(_trustedForwarder) {
+    constructor(address _trustedForwarder, IERC20 _nativeToken, address _rewardsDistribution ) ERC2771Context(_trustedForwarder) {
```
Comment (QEDK) [inline contracts/contracts/OpenSignal.sol:52]: Where is `_rewardsDistribution` used?

**Review by QEDK — CHANGES_REQUESTED** [review]
Can you use run solhint on the code, thx

File: contracts/contracts/OpenSignal.sol (hunk 1 above)
Comment (Amirjab21) [reply contracts/contracts/OpenSignal.sol:52]: It's used to get instance of rewardDistribution which is then called everytime a token is staked/unstaked.

Comment (QEDK) [reply contracts/contracts/OpenSignal.sol:52]: You're making the instance on line 58, that is what is used in the contract, both approaches are fine but currently this variable is not in use

Comment (Amirjab21) [reply contracts/contracts/OpenSignal.sol:52]: Ah yep, this is now resolved. I changed it so that we pass in the address of the already deployed contract.

**Review by QEDK — APPROVED** [review]
LGTM!

**Pull request merged**